cargo-pgrx/src/command/init.rs

```diff
@@ -271,14 +271,35 @@ fn untar(bytes: &[u8], pgrxdir: &PathBuf, pg_config: &PgConfig) -> eyre::Result<
         .wrap_err("failed to spawn `tar`")?;
 
     let stdin = child.stdin.as_mut().expect("failed to get `tar`'s stdin");
-    stdin.write_all(bytes)?;
-    stdin.flush()?;
+    let mut error = None;
+
+    // In order to capture and report any output from the child's stderr (in the event of an error),
+    // we need to wait until `.wait_with_output()` is called. Bailing early here via `?` or `.expect()`
+    // means only the immediate error is reported, such as "Broken Pipe". The same applies for the subsequent
+    // `.flush()` call immediately after.
+    match stdin.write_all(bytes) {
+        Ok(_) => (),
+        Err(e) => error = Some(e),
+    }
+
+    match stdin.flush() {
+        Ok(_) => (),
+        Err(e) => error = Some(e),
+    }
```

Comment:
Is it possible for both of these to error? We'd hypothetically be dumping one error on the floor in favor of another, again.

Reply:
I would be very surprised if flush did anything on this type. But we should probably either take the first error or use chaining, if we care.

Reply:
I thought about this as I was writing it, but I wasn't entirely sure how to handle it without making it "shift right" or making a plate of spaghetti out of it.

Reply:
You will see me tend to incrementally construct and then immediately pattern-match on tuples a lot in my code for this reason.